src/indexing/transforms/batch_processor.py

```
@@ -0,0 +1,360 @@
+    def _get_batch_client(self) -> BatchAPIClient:
+        """Get or create Batch API client."""
+        if self._batch_client is None:
+            import os
+
+            from openai import OpenAI
+
+            api_key = os.getenv("OPENAI_API_KEY")
+            if not api_key:
+                raise RuntimeError(
+                    "OPENAI_API_KEY not set. Required for Batch API."
+                )
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)